### DISCUSSION TITLE
Meanings of error codes.

### DISCUSSION THREAD

**Discussion by dgsdestinygamestudios** [GodotNuts/GodotFirebase#367] [Q&A]
I've been looking through the code but I'm having a hard time understanding what error code means what.

[Firebase Error] >> Action in error was: 0
[Firebase Error] >> Unknown error

Like here. I'd appreciate if anyone explains these error codes.

**Comment by WolfgangSenff** [comment]
Yeah, I have a ticket to make a better explanation for those. I'll get around to it at some point. It looks like you're likely getting an error that we can't really help with without actually seeing your project possibly. https://github.com/GodotNuts/GodotFirebase/blob/7b90285c200353f370556b26975db287bb8a1456/addons/godot-firebase/firestore/firestore_task.gd#L174

This is where that generic error comes up. What are you trying to do when that error occurs?

**Reply by dgsdestinygamestudios** [reply]
I'm trying to get a dictionary from an array per X seconds. Most of the time i get this error. Some other times it works as intended. That's what confused me.

**Comment by WolfgangSenff** [comment]
Generally speaking, I don't think you'd want to do that - you're using Firestore? You're going to have a massive bill if you keep on with it. Firestore is not for repeated things like that, because it requires two round-trips that can be expensive in order to get the data. If you're really trying something like that, I'd recommend the realtime database instead. You can't use arrays (in the strictest sense) with it, but you get array-like things and it's always up-to-date, so you don't have to constantly ping in order to get it again; it will just always be what you want eventually.

**Reply by dgsdestinygamestudios** [reply]
It's not that important thing for me the one I'm trying to achieve. I'm aware of realtime database, I'm just messing around with firestore. Thank you for your answers.

**Comment by dgsdestinygamestudios** [comment]
Though I need help on retrieving data from realtime database. You've explained pushing data but not pulling.

**Reply by WolfgangSenff** [reply]
To pull it, you just make a reference to the path you're looking at and connect to new_data_update. It's all on the Wiki.

**Reply by dgsdestinygamestudios** [reply]
It'll fire if data is updated if I'm not looking at it incorrectly. What if i want to retrieve regardless of change?

**Reply by WolfgangSenff** [reply]
You're looking at it incorrectly. It pulls back the entire dataset that's at that point the first time, and then updates every time thereafter. There *is* a proposal though for a "oneshot" pull of everything that I'm considering implementing, just don't have the time right now. Feel free to look for it in the issues (it's very recent) and upvote it and that may help.
